simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileLargeDataSets.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.simulator.generator;
+
+import static com.hedera.block.simulator.generator.Utils.readFileBytes;
+import static java.lang.System.Logger.Level.INFO;
+
+import com.hedera.block.simulator.config.data.BlockStreamConfig;
+import com.hedera.block.simulator.config.types.GenerationMode;
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.IOException;
+import javax.inject.Inject;
+
+/** A block stream manager that reads blocks from files in a directory. */
+public class BlockAsFileLargeDataSets implements BlockStreamManager {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    private final String blockstreamPath;
+    private int currentBlockIndex = 0;
+    private int currentBlockItemIndex = 0;
+
+    private Block currentBlock = null;
+    private final int paddedLength;
+    private final String fileExtension;
+
+    /**
+     * Constructs a new BlockAsFileLargeDataSets instance.
+     *
+     * @param config the block stream configuration
+     */
+    @Inject
+    public BlockAsFileLargeDataSets(@NonNull BlockStreamConfig config) {
```

Comment:
I think we need to discuss this as a team, currently we are relaying on @NonNull and static analysis to find potential null bugs, I would say that is a trade off for better performance overall or maybe simplicity.

Reply:
The annotation is only ever useful at compile time, and is not processed at runtime.  It shouldn't even be retained at runtime, as it's is exclusively intended for static analysis tooling.
The general recommendation (and what Services does) is to use Objects.requireNonNull
(e.g. `this.blockstreamPath = Objects.requireNonNull(config).folderRootPath();`) on the first use of any parameter so annotated.

That said, `Objects.requireNonNull` doesn't do anything magic, it just throws a `NullPointerException`, so if the first use of the parameter would throw that exception (as is the case here), then adding the additional call is just wasted processing.

Reply:
Noted, let's discuss this so we can be on the same page about this, cause I would generally always add the precondition, and we will always be commenting each other about this.

Reply:
@jsync-swirlds I agree with you that in general we should always use these preconditions, in some frameworks, there is runtime processing of these annotations (like Spring) and I was wondering if there is such a thing here. Since there is not, failing fast is a good idea.